### PR TITLE
174. Adding support for mTLS DN logging to ALS

### LINKED ISSUE #174: Store mTLS certidicates in ALS
Pick certificates DN in dispatcher, transfer it as header to afcserver/webui,
Add table to ALS for it, learn Alembic in process

### PR DESCRIPTION
**Pull request by FedorSherstyukBroadcom**
- HTTPS-related parameters added to afc_load_tool.py
- mTLS-DN proxy header added to dispatcher configuration to marshal mTLS DN to afcserver/msghnd
- mtls_dn parameter added to als.als_afc_request()
- afcserver/msghnd drain content of received mTLSD-DN header to als.als_afc_request() via mtls_dn parameter
- Alembic for ALS database introduced to add mtls_dn table to ALS database on the fly
- mtls_dn tqable added to database scheme (ALS.dbml) and all its numerous derivatives to facilitate mtlks_dn table creation when database created from scratch
- Alembic support added to als_siphon.py for automatic upgrade ALS  database scheme
- Support for mtls_dn table added to als_siphon.py to facilitate putting received mTLS DN to it
- Support for DN output and CN filtering added to als_query.py

Tested with HTTP, HTTPS, HTTPS/mTLS modes - works as expected

Closes #174

### CODE REVIEW

File: als/Dockerfile.siphon
Comment (vitalypetrovisbest) [inline als/Dockerfile.siphon:1]: I'm not against, but what is the purpose of this?

File: dispatcher/nginx.conf.template [hunk 1]
```diff
@@ -117,6 +117,7 @@ http {
             proxy_set_header X-Real-IP $remote_addr;
             proxy_pass http://webui;
             proxy_redirect http:// $scheme://;
+            proxy_set_header "mTLS-DN" '$ssl_client_s_dn';
```
Comment (vitalypetrovisbest) [inline dispatcher/nginx.conf.template:120]: should we, maybe, put this under some "if afc_mtls_enforce" But be careful, ifs in  nginx configs are very tricky

File: dispatcher/nginx.conf.template [hunk 2]
```diff
@@ -136,6 +137,7 @@ http {
             # transmit a request to a proxied server
             proxy_send_timeout 720;
             proxy_pass http://msghnd$uri$is_args$args;
+            proxy_set_header "mTLS-DN" '$ssl_client_s_dn';
```
Comment (vitalypetrovisbest) [inline dispatcher/nginx.conf.template:140]: ditto

File: als/als_siphon.py [hunk 3]
```diff
@@ -2736,11 +2794,14 @@ class AfcMessageTableUpdater(TableUpdaterBase[int, AlsMessageBundle]):
     _col_tx_time          -- AFC Response timetag column
     _rx_envelope_digest   -- AFC Request envelope digest column
     _tx_envelope_digest   -- AFC Response envelope digest column
+    _tx_envelope_digest   -- AFC Response envelope digest column
```
Comment (vitalypetrovisbest) [inline als/als_siphon.py:2797]: duplication

**Review by vitalypetrovisbest — COMMENTED** [review]
Okay in general. Just couple of typos and one explanation request. and discussion regarding dispatcher config.

File: dispatcher/nginx.conf.template (hunk 1 above)
Comment (vitalypetrovisbest) [reply dispatcher/nginx.conf.template:120]: resolved in a call - no problems during testing

File: dispatcher/nginx.conf.template (hunk 2 above)
Comment (vitalypetrovisbest) [reply dispatcher/nginx.conf.template:140]: ditto

File: als/ALS.sql [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2022 Broadcom. All rights reserved.
- * The term "Broadcom" refers solely to the Broadcom Inc. corporate affiliate
+ * The term  refers solely to the Broadcom Inc. corporate affiliate
```
[the remark this answers is not in the document]
Comment (FedorSherstyukBroadcom) [reply als/ALS.sql:3]: Windows awk glitch. Will redo on Linux side

**Review by vitalypetrovisbest — DISMISSED** [review]

File: als/Dockerfile.siphon
Comment (FedorSherstyukBroadcom) [reply als/Dockerfile.siphon:1]: Purpose of what?

Comment (FedorSherstyukBroadcom) [reply als/Dockerfile.siphon:1]: This was just a way to make literal a named constant. Will add comment that this value is not intended for external modification.

File: als/ALS.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (FedorSherstyukBroadcom) [reply als/ALS.sql:3]: Well, it's not awk, it's me. Fixed

**Review by vitalypetrovisbest — APPROVED** [review]

**Pull request merged**